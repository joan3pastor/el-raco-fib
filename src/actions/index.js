export const GET_TOKEN = 'GET_TOKEN';
export const GET_CODE = 'GET_CODE';
export const LOAD_TOKEN = 'LOAD_TOKEN';
export const DELETE_TOKEN = 'DELETE_TOKEN';
export const HAS_CONNEXION = 'HAS_CONNEXION';
export const AVISOS = 'AVISOS';
export const HORARI = 'HORARI';
export const HORARI_MODIFICAT = 'HORARI_MODIFICAT';
export const ASSIG = 'ASSIG';
export const NOTICIES = 'NOTICIES';
export const IMGLABS = 'IMGLABS';
export const STORAGEAVISOS = 'STORAGEAVISOS';
export const EXAMENS = 'EXAMENS';
export const EVENTS = 'EVENTS';
export const PRACTIQUES = 'PRACTIQUES'; 
export const SETTINGS = 'SETTINGS';
export const NOTES = 'NOTES';
export const LANG = 'LANG';

import _ from 'lodash';
import AsyncStorage from '@react-native-community/async-storage';

// Languages pack
import lang from '../languages/index'

export function getCode(url) {
    const codeIndex = url.indexOf('code=') + 5;
    if (codeIndex > 4) {
        const code = url.substring(codeIndex);
        return {
            type: GET_CODE,
            payload: { code, code_error: '' }
        };
    }
    else {
        return {
            type: GET_CODE,
            payload: { code: '', code_error: 'Accés denegat' }
        };
    }
}

export function getToken(token) {
    // Esta función es llmada una vez se consigue el response debido 
    // a que es complicado trabajar con promesas en action creators...
    return {
        type: GET_TOKEN,
        resp: token
    };
}

export function tokenToRedux(token) {
    if (_.isEmpty(token)) {
        return {
            type: LOAD_TOKEN,
            resp: {token: ''}
        }
    }
    return {
        type: LOAD_TOKEN,
        resp: token
    };
}

export function deleteToken() {
    return {
        type: DELETE_TOKEN,
        resp: {token: ''}
    }
}

export function updateConnexion(b) {
    return {
        type: HAS_CONNEXION,
        payload: b
    }
}

export function avisosToRedux(resp) {
    return {
        type: AVISOS,
        resp: resp
    }
}

export function horariToRedux(resp) {
    return {
        type: HORARI,
        resp: resp
    }
}

export function horariModificatToRedux(respHorari, modhorari) {

    // anadir clases creadas
    var horari = respHorari;

    modhorari.afegits.forEach((clase) => horari.results.push(clase));

    // eliminar clases borradas
    var indicesAEliminar = [];
    modhorari.eliminats.forEach((c1) => {
        horari.results.forEach((c2, index) => {
            if (c1.dia_setmana == c2.dia_setmana && c1.codi_assig == c2.codi_assig && c1.inici == c2.inici && c1.grup == c2.grup && c1.durada == c2.durada && c1.aules == c2.aules) {
                indicesAEliminar.push(index);
            }
        })
    })

    indicesAEliminar.sort().reverse();
    indicesAEliminar.forEach((index) => {
        horari.results.splice(index, 1);
    });


    return {
        type: HORARI_MODIFICAT,
        resp: horari // resp = {count:1, results:[]}
    }
}

export function assigToRedux(resp) {
    return {
        type: ASSIG,
        resp: resp
    }
}

export function noticiesToRedux(resp) {
    return {
        type: NOTICIES,
        resp: resp
    }
}

export function imglabsToRedux(resp) {
    return {
        type: IMGLABS,
        resp: resp
    }
}

export function storageAvisosToRedux(data) {
    return {
        type: STORAGEAVISOS,
        payload: data
    }
}

export function examensToRedux(resp) {
    return {
        type: EXAMENS,
        resp: resp
    }
}

export function eventsToRedux(resp) {
    return {
        type: EVENTS,
        resp: resp
    }
}

export function practiquesToRedux(resp) {
    return {
        type: PRACTIQUES,
        resp: resp
    }
}

export function settingsToRedux(data) {
    return {
        type: SETTINGS,
        resp: data
    }
}

export function notesToRedux(data) {

    AsyncStorage.setItem("notes", JSON.stringify(data));

    return {
        type: NOTES,
        resp: data
    }
}

export async function loadLanguage() {

    var langSelected = "ca";

    const langUnparsed = await AsyncStorage.getItem("lang");
    if (langUnparsed != "" && langUnparsed != null && langUnparsed != undefined) {
        langSelected = JSON.parse(langUnparsed)
    } else {
        AsyncStorage.setItem("lang", JSON.stringify(langSelected));
    }

    const langPack = lang[langSelected];

    return {
        type: LANG,
        resp: langPack
    }
}

export function changeLanguage(data) {

    AsyncStorage.setItem("lang", JSON.stringify(data));
    const langPack = lang[data];

    return {
        type: LANG,
        resp: langPack
    }
}
