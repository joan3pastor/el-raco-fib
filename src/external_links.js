// OAUTH:
export const authURL = 'https://api.fib.upc.edu/v2/o/authorize';
export const tokenURL = 'https://api.fib.upc.edu/v2/o/token';
export const clientID = '8IVfHurpzfHhC68LdfpIoRoh7uWcr5bXBkCGroV4';
export const clientSECRET = '8W3Xbf3kRfIQK3rW8QR2xED3TxZ1cI3kMv4VnqyeoreruGNKXBnL8MtcJUztndI7QiuqvJhkvfNE53fOSfbqOT4DEH75d2TmgrmPTHeU5HDuRrA2tFlyYSBoKH31O9K4';
export const callbackURI = 'apifib://callback'
export const revokeURL = "https://api.fib.upc.edu/v2/o/revoke_token/";

// API URLs:
export const joURL = 'https://api.fib.upc.edu/v2/jo/';
export const fotoURL = 'https://api.fib.upc.edu/v2/jo/foto.jpg';
export const avisosURL = 'https://api.fib.upc.edu/v2/jo/avisos/';
export const horariURL = 'https://api.fib.upc.edu/v2/jo/classes/';
export const assigURL = 'https://api.fib.upc.edu/v2/jo/assignatures/';
export const noticiesURL = 'https://api.fib.upc.edu/v2/noticies/';
export const imglabsURL = 'https://api.fib.upc.edu/v2/laboratoris/imatges/';
export const eventsURL = 'https://api.fib.upc.edu/v2/events/';
export const practiquesURL = 'https://api.fib.upc.edu/v2/jo/practiques/';
export const placesURL = 'https://api.fib.upc.edu/v2/assignatures/places/';
export const mobilitatURL = 'https://api.fib.upc.edu/v2/universitats-erasmus/';
export const lecturesURL = 'https://api.fib.upc.edu/v2/lectures/properes/';
export const QActualURL = 'https://api.fib.upc.edu/v2/quadrimestres/actual/';
export const aulesURL = 'https://api.fib.upc.edu/v2/aules/';

// UPC Notícies RSS Channel
export const upcNoticiesRSSca = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSes = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";
export const upcNoticiesRSSen = "https://www.upc.edu/ca/sala-de-premsa/noticies/RSS";

// Reservas Bibl. 
export const salesBiblioteca = 'https://apps.bibliotecnica.upc.edu/reserva_sales/comunes/disponibilidad.php?Id_centro=1&Admin=1&Parametros=TnVtX2NvbHVtbmFzPTYmQWx0bz0mQW5jaG89MTAwJSZGdWVudGU9QXJpYWwmVGFtYW5vX2Z1ZW50ZT0xOHB4JlRhbWFub19jb250b3Jubz0ycHgmTWFyZ2VuX2RvY3VtZW50bz0wcHgmQ29sb3JfZm9uZG89ZmZmZmZmJlBhZGRpbmdfY2VsZGE9MTBweCZNb3N0cmFyX3RpdHVsbz0xJkFicmV2aWF0dXJhcz0xJk1vc3RyYXJfbGlua19hZG1pbj0wJk1vc3RyYXJfdGV4dG9fZGlzcG9uaWJsZXM9MCZDb21wYWN0YXJfdWJpY2FjaW9uZXM9MCZNb3N0cmFyX2VubGFjZV9pbmZvcm1hY2lvbj0wJk1vc3RyYXJfYm90b25fcmVzZXJ2YXI9MQ==';

// Enllaços d'interès
export const linkDEFIB = 'http://defib.upc.edu/qui_som';
export const linkAssociacions = 'https://www.fib.upc.edu/ca/la-fib/vida-universitaria/associacions';
export const linkEduroam = 'https://serveistic.upc.edu/ca/wifiupc/documentacio/assistent-de-configuracio-per-eduroam';
export const linkCorreu = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/correu-destudiant';
export const linkImpressio = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/servei-dimpressio';
export const linkUPClink = 'https://www.fib.upc.edu/ca/la-fib/serveis-tic/vpn-upclink';
export const linkAcordsUPC = 'https://serveistic.upc.edu/ca/distsoft/el-servei/programari-per-a-estudiantsi-professors';
export const linkVMware = 'https://raco.fib.upc.edu/vmware';
export const linkGithub = 'https://education.github.com/pack';
export const linkAzure = 'https://azure.microsoft.com/en-us/free/students/';
export const linkOffice = 'https://www.microsoft.com/es-es/education/products/office';
export const linkGCP = 'https://cloud.google.com/edu/';

// Feedback
export const playstorelink = 'https://play.google.com/store/apps/details?id=com.fibupc';

// Programes de Mobilitat
export const mobInfoGeneral = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/estades-academiques';
export const mobSessionsInfo = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/sessions-informatives';
export const mobCalendari = 'https://www.fib.upc.edu/ca/mobilitat/outgoing/calendari-mobilitat';
export const mobBeques = 'https://www.upc.edu/sga/ca/Beques/BequesMobilitat';

// Github
export const githubURL = "https://github.com/joan3pastor/el-raco-fib";
