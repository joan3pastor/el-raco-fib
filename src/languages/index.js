import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};