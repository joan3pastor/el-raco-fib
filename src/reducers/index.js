import { combineReducers } from "redux";
import reducer_token from "./reducer_token";
import reducer_code from "./reducer_code";
import reducer_code_error from "./reducer_code_error";
import reducer_has_connexion from './reducer_has_connexion';
import reducer_avisos from './reducer_avisos';
import reducer_horari from './reducer_horari';
import reducer_horari_modificat from './reducer_horari_modificat';
import reducer_assig from './reducer_assig';
import reducer_noticies from './reducer_noticies';
import reducer_imglabs from "./reducer_imglabs";
import reducer_storage_avisos from "./reducer_storage_avisos";
import reducer_examens from './reducer_examens';
import reducer_events from './reducer_events';
import reducer_practiques from './reducer_practiques';
import reducer_settings from './reducer_settings';
import reducer_notes from './reducer_notes';
import reducer_lang from "./reducer_lang";

const rootReducer = combineReducers({
  token: reducer_token,
  code: reducer_code, //No siempre estÃ¡. Solo tiene valor si entra por primera vez. Para conseguir codigo -> SyncStorage
  code_error: reducer_code_error,
  hasConnexion: reducer_has_connexion,
  settings: reducer_settings,

  avisos: reducer_avisos,
  storageAvisos: reducer_storage_avisos,
  horari: reducer_horari,
  horari_modificat: reducer_horari_modificat,
  assig: reducer_assig,
  noticies: reducer_noticies,
  imglabs: reducer_imglabs,

  examens: reducer_examens,
  events: reducer_events,
  practiques: reducer_practiques,

  notes: reducer_notes,

  lang: reducer_lang,
  
});

export default rootReducer;

