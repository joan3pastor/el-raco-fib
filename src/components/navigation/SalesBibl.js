import React, { Component } from 'react';
import { View } from 'react-native';
import { WebView } from 'react-native-webview';
import { connect } from 'react-redux';
import _ from 'lodash';
import {salesBiblioteca} from "../../external_links";

import {  } from '../../actions';

class SalesBibl extends Component {

  render() {
    return (
      <View style={{flex:1, backgroundColor:'#fafafa', margin:2, marginTop:0}}>
        <WebView textZoom={90}
          scalesPageToFit={false}
          source={{uri: salesBiblioteca}}
        />
      </View>
    );
  }
}

function mapStateToProps(state) {
  return { 
    
  };
}

export default connect(mapStateToProps, {  })(SalesBibl);
