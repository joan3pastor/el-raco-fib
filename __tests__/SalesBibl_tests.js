import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import axios from 'axios'

import SalesBibl from "../src/components/navigation/SalesBibl";
import {salesBiblioteca} from "../src/external_links";

Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {};

beforeEach( () => {})

describe('Testing SalesBibl component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <SalesBibl store={mockStore(initialState)} token={undefined} />
    );
    expect(wrapper.dive().dive()).toMatchSnapshot();
  });
  it("sales bibl. link works", async () => {
    resp = await axios.get(salesBiblioteca);
    expect(resp.status == 200).toBe(true);
  });
});

