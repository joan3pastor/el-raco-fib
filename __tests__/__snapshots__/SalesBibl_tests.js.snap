// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SalesBibl component renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "margin": 2,
      "marginTop": 0,
    }
  }
>
  <WebView
    cacheEnabled={true}
    originWhitelist={
      Array [
        "http://*",
        "https://*",
      ]
    }
    scalesPageToFit={false}
    source={
      Object {
        "uri": "https://apps.bibliotecnica.upc.edu/reserva_sales/comunes/disponibilidad.php?Id_centro=1&Admin=1&Parametros=TnVtX2NvbHVtbmFzPTYmQWx0bz0mQW5jaG89MTAwJSZGdWVudGU9QXJpYWwmVGFtYW5vX2Z1ZW50ZT0xOHB4JlRhbWFub19jb250b3Jubz0ycHgmTWFyZ2VuX2RvY3VtZW50bz0wcHgmQ29sb3JfZm9uZG89ZmZmZmZmJlBhZGRpbmdfY2VsZGE9MTBweCZNb3N0cmFyX3RpdHVsbz0xJkFicmV2aWF0dXJhcz0xJk1vc3RyYXJfbGlua19hZG1pbj0wJk1vc3RyYXJfdGV4dG9fZGlzcG9uaWJsZXM9MCZDb21wYWN0YXJfdWJpY2FjaW9uZXM9MCZNb3N0cmFyX2VubGFjZV9pbmZvcm1hY2lvbj0wJk1vc3RyYXJfYm90b25fcmVzZXJ2YXI9MQ==",
      }
    }
    textZoom={90}
    useSharedProcessPool={true}
    useWebKit={true}
  />
</View>
`;
